src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -278,8 +278,28 @@ bool TokenizeSnippets::AppropriateForSubgraph(const std::shared_ptr<const Node>&
 TokenizeSnippets::TokenizeSnippets(const TokenizationConfig& config) {
     MATCHER_SCOPE(TokenizeSnippets);
 
-    auto label = ov::pass::pattern::any_input([](const ov::Output<ov::Node>& out) {
+    auto label = ov::pass::pattern::any_input([config](const ov::Output<ov::Node>& out) {
         const auto n = out.get_node_shared_ptr();
+        // Config-aware gating: optionally reject specific Transpose cases around MatMul
+        if (ov::is_type<ov::op::v1::Transpose>(n)) {
```

Comment:
Let's move this check to `is_supported_transpose` lambda in `is_supported_op`. We will need to change their signatures, but I think it is okay